qubesbuilder/plugins/source_rpm/scripts/prepare-chroot-base

```diff
@@ -0,0 +1,237 @@
+
+        mkdir -p -- "${DOWNLOAD_DIR}"
+        yumconf=$(mktemp)
+
+        if [ "x${FEDORA_MIRROR}" != "x" ]; then
+            awk '
+            BEGIN {
+                mirror=ARGV[1];     delete ARGV[1];
+                releasever=ARGV[2]; delete ARGV[2];
+            }
+            {
+                gsub("^metalink", "#metalink");
+                gsub("^#baseurl=.*/(linux|fedora)/", "baseurl=" mirror "/");
+                gsub("\\$releasever", releasever);
+                print;
+            }' "${FEDORA_MIRROR%/}" "${DIST_VER}" \
+                <"${DNF_BOOTSTRAP_CONF}" \
+                >"$yumconf"
+        else
+            sed -e "s/\\\$releasever/${DIST_VER}/g" \
+                -e "s#\\\$plugindir#${PLUGIN_DIR}#g" \
+                <"${DNF_BOOTSTRAP_CONF}" \
+                >"$yumconf"
+        fi
+    fi
+
+    if [ "${DIST_NAME}" = "centos-stream" ]; then
+        INITIAL_PACKAGES=(filesystem setup epel-release)
+
+        # coreutils conflicts with coreutils-single
+        INITIAL_PACKAGES=(--exclude=coreutils-single "${INITIAL_PACKAGES[@]}")
+        INITIAL_PACKAGES+=(centos-stream-release centos-stream-repos dnf dnf-plugins-core)
+
+        # Add groupadd, su etc.
+        INITIAL_PACKAGES+=(shadow-utils util-linux)
+
+        mkdir -p -- "${DOWNLOAD_DIR}"
+        yumconf=$(mktemp)
+        yumconftmp=$(mktemp)
+
+        sed -e "s/\\\$releasever/${DIST_VER}/g" \
+            <"${DNF_BOOTSTRAP_CONF}" \
+            >"$yumconf"
+
+        if [ "x${CENTOS_MIRROR}" != "x" ]; then
+            awk '
+            BEGIN {
+                mirror=ARGV[1];     delete ARGV[1];
+            }
+            {
+                gsub("^mirrorlist", "#mirrorlist");
+                gsub("^#baseurl=.*/(centos|CentOS)/", "baseurl=" mirror "/");
+                print;
+            }' "${CENTOS_MIRROR%/}" \
+                <"$yumconf" \
+                >"$yumconftmp"
+            mv "$yumconftmp" "$yumconf"
+        fi
+
+        if [ "x${EPEL_MIRROR}" != "x" ]; then
+            awk '
+            BEGIN {
+                mirror=ARGV[1];     delete ARGV[1];
+            }
+            {
+                gsub("^metalink", "#metalink");
+                gsub("^#baseurl=.*/epel/", "baseurl=" mirror "/");
+                print;
+            }' "${EPEL_MIRROR%/}" \
+                <"$yumconf" \
+                >"$yumconftmp"
+            mv "$yumconftmp" "$yumconf"
+        fi
+    fi
+
+    "$DNF" "${DNF_OPTS[@]}" -c "$yumconf" -y install "${INITIAL_PACKAGES[@]}"
+    rm -f "$yumconf"
+
+    echo "INFO: Verifying signatures..."
+    set +x
+    for file in "${DOWNLOAD_DIR}"/*; do
+        result=$(rpmkeys "${RPM_OPTS[@]}" --root="${INSTALL_DIR}" --checksig -- "${file}") || {
+            echo "Filename: ${file} failed verification.  Exiting!"
+            exit 1
+        }
+        [[ "$result" == "$file: digests signatures OK" ]] || {
+            echo "Filename: ${file} is not signed.  Exiting!"
+            exit 1
+        }
```

Comment:
```suggestion
        result=$(LC_ALL=C rpmkeys "${RPM_OPTS[@]}" --root="${INSTALL_DIR}" --checksig - < "${file}") || {
            echo "Filename: ${file} failed verification.  Exiting!"
            exit 1
        }
        [[ "$result" == "-: digests signatures OK" ]] || {
            echo "Filename: ${file} is not signed.  Exiting!"
            exit 1
        }
```

The output of `rpmkeys -K` is locale-dependent, and using `-` for `stdin` allows not worrying about the file name.